crates/context/config/src/client/env/proxy/query/context_storage_entries.rs

```diff
@@ -0,0 +1,105 @@
+use candid::{Decode, Encode};
+use serde::Serialize;
+use starknet::core::codec::{Decode as StarknetDecode, Encode as StarknetEncode};
+use starknet_crypto::Felt;
+
+use crate::client::env::proxy::starknet::{
+    CallData, ContextStorageEntriesResponse, StarknetContextStorageEntriesRequest,
+};
+use crate::client::env::Method;
+use crate::client::protocol::icp::Icp;
+use crate::client::protocol::near::Near;
+use crate::client::protocol::starknet::Starknet;
+use crate::types::ContextStorageEntry;
+
+#[derive(Clone, Debug, Serialize)]
+pub(super) struct ContextStorageEntriesRequest {
+    pub(super) offset: usize,
+    pub(super) limit: usize,
+}
+
+impl Method<Near> for ContextStorageEntriesRequest {
+    const METHOD: &'static str = "context_storage_entries";
+
+    type Returns = Vec<ContextStorageEntry>;
+
+    fn encode(self) -> eyre::Result<Vec<u8>> {
+        serde_json::to_vec(&self).map_err(Into::into)
+    }
+
+    fn decode(response: Vec<u8>) -> eyre::Result<Self::Returns> {
+        // Decode the response as Vec of tuples with boxed slices
+        let entries: Vec<(Box<[u8]>, Box<[u8]>)> = serde_json::from_slice(&response)
+            .map_err(|e| eyre::eyre!("Failed to decode response: {}", e))?;
+
+        // Convert to ContextStorageEntry
+        Ok(entries
+            .into_iter()
+            .map(|(key, value)| ContextStorageEntry {
+                key: key.to_vec(),
+                value: value.to_vec(),
+            })
```

Comment:
`[u8]::to_vec` copies the slice.

```suggestion
        let entries: Vec<(Vec<u8>, Vec<u8>)> = serde_json::from_slice(&response)
            .map_err(|e| eyre::eyre!("Failed to decode response: {}", e))?;

        // Convert to ContextStorageEntry
        Ok(entries
            .into_iter()
            .map(|(key, value)| ContextStorageEntry {
                key,
                value,
            })
```

or

```suggestion
        let entries: Vec<(Box<[u8]>, Box<[u8]>)> = serde_json::from_slice(&response)
            .map_err(|e| eyre::eyre!("Failed to decode response: {}", e))?;

        // Convert to ContextStorageEntry
        Ok(entries
            .into_iter()
            .map(|(key, value)| ContextStorageEntry {
                key: key.into(),
                value: value.into(),
            })
```